4b/Grains/src/Test1.cs

```diff
@@ -11,19 +9,22 @@ public class Test1 : Orleans.Grain, ITest1
     {
         public async Task<TestDetails> HelloWorldTest(string testListId)
         {
-            var executionTimeDetails = new UnitTestExecutionTime();
-            var stopwatch = new Stopwatch();
+            var startTime = DateTime.UtcNow;
+            await Task.Delay(200);
+            var endTime = DateTime.UtcNow;
+            var duration = endTime - startTime;
 
-            executionTimeDetails.StartTime = DateTime.UtcNow.ToString(CultureInfo.InvariantCulture);
-            stopwatch.Start();
-            
-            await Task.Delay(100);
-
-            stopwatch.Stop();
-            executionTimeDetails.EndTime = DateTime.UtcNow.ToString(CultureInfo.InvariantCulture);
-            executionTimeDetails.Duration = stopwatch.Elapsed.ToString();
-            
-            var unitTest = Helpers.UnitTestCreator(this.GetType(), Helpers.CallerName(), testListId, executionTimeDetails);
+            var unitTest = Helpers.TestDetailsCreator(new TestCreatorParameters()
+            {
+                ClassType = GetType(),
+                CallerName = Helpers.CallerName(),
+                TestListId = testListId,
+                StartTime = startTime.ToString("yyyy-MM-ddThh:mm:ss.fff"),
+                EndTime = endTime.ToString("yyyy-MM-ddThh:mm:ss.fff"),
```

Comment:
Was this format not supposed to be a constant string?

Reply:
Oh! One of us can extract these first thing when we go back to working on these. I'll leave a comment on my local branch. Thanks for pointing it out :)